src/array/cuda/csr_mm.cu

```diff
@@ -110,7 +190,7 @@ std::pair<CSRMatrix, NDArray> CusparseSpgemm(
   // copy the final products to the matrix C
   CUSPARSE_CALL(cusparseSpGEMM_copy(
       thr_entry->cusparse_handle, transA, transB, &alpha, matA, matB, &beta,
-      matC, dtype, CUSPARSE_SPGEMM_DEFAULT, spgemmDesc));
+      matC, dtype, alg, spgemmDesc));
 
   device->FreeWorkspace(ctx, workspace1);
```

Comment:
Can we free `workspace1` before the actual computing?

Reply:
The total number of products in the SpGEMM and some other internal information are stored in `workspace1`. So, we cannot free `workspace1` until the end of SpGEMM computation.